### DISCUSSION TITLE
Trivy Secret Scanning does not detect Github fine-grained tokens

### DISCUSSION THREAD

**Discussion by dn-scribe** [aquasecurity/trivy#5649] [Bugs]
### Description

Github fine grained tokens' format differ from the classical tokens.
See demo of the problme [here](https://github.com/scribe-security/secret-experiment/security/code-scanning/1)

Here is a screen shot:
![image](https://github.com/aquasecurity/trivy/assets/85332941/6116dcc5-7a1e-435f-9c33-18ae818477f2)

### Desired Behavior

detect such secrets.

### Actual Behavior

does not detect.

### Reproduction Steps

```bash
1.See here: https://github.com/scribe-security/secret-experiment.
```

### Target

Git Repository

### Scanner

Secret

### Output Format

None

### Mode

None

### Debug Output

```bash
Not needed.
```

### Operating System

Github runner (ubuntu)

### Version

```bash
latest
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @dn-scribe 
Thanks for your report!

Created #5650 for this task.

Regards, Dmitriy
